pzsd_bot/cogs/points.py

```diff
@@ -200,18 +222,32 @@ async def fetch_leaderboard(
                 .where(*args)
                 .group_by(pzsd_user.c.id)
             )
-            leaderboard = sorted(result.fetchall(), key=lambda r: r.sum, reverse=True)
+            sorted_points = sorted(result.fetchall(), key=lambda r: r.sum, reverse=True)
+
+        logger.info("Leaderboard length is %s", len(sorted_points))
+
+        leaderboard = [
+            (rank, name, points) for rank, (name, points) in enumerate(sorted_points, 1)
+        ]
+
+        if paginate:
+            iterator = iter(leaderboard)
+            leaderboard = []
+            while chunk := tuple(islice(iterator, page_size)):
```

Comment:
What version of python are you using? 3.12 gives you https://docs.python.org/3/library/itertools.html#itertools.batched

Reply:
lol yea I stole this recipe from there. We're on 3.10 and I was being lazy. I suppose I should just update your VPS to 3.12